src/amdcert.h

```diff
@@ -49,14 +49,18 @@ class AMDCert {
 private:
     SEVDevice *m_sev_device;
     SEV_ERROR_CODE amd_cert_validate_sig(const amd_cert *cert,
-                                         const amd_cert *parent);
+                                         const amd_cert *parent,
+                                         ePSP_DEVICE_TYPE device_type);
     SEV_ERROR_CODE amd_cert_validate_common(const amd_cert *cert);
     bool usage_is_valid(AMD_SIG_USAGE usage);
     SEV_ERROR_CODE amd_cert_validate(const amd_cert *cert,
                                      const amd_cert *parent,
-                                     AMD_SIG_USAGE expected_usage);
+                                     AMD_SIG_USAGE expected_usage,
+                                     ePSP_DEVICE_TYPE device_type);
     SEV_ERROR_CODE amd_cert_public_key_hash(const amd_cert *cert,
                                             hmac_sha_256 *hash);
+    // Retrieves information on device type (naples/rome/milan...) based on key id
+    ePSP_DEVICE_TYPE get_device_type(const amd_cert *ark);
```

Comment:
Wouldn't get_device_type() need to eventually be used by more than just the AMDCert class? What was the want for moving it out of the sevcore area?

Reply:
During the certificate chain validation, `get_device_type` is used to differentiate between the different hardware generations and thus AMD root keys. If the origin of the certificates is not the platform also which is validating them, then obtaining the device_type from the validating platform's firmware could lead to wrong results as a different generation of ARKs/ASKs could be used. In my case, the validation failed completely since I used the tool on a non-SEV device. As a solution, I took the information provided in the ARK certificate to deduce the correct hardware generation.
By doing that, the get_device_type is no longer OS-dependent, which is why I moved it into the AMDCert class.
If a function wants to obtain the device_type of the given host platform, an additional function obtaining these information would make sense, you're right.